TestDeletingRunningPod is flakey

```
    --- FAIL: TestFunctionalSuite/TestDeletingRunningPod (9.56s)
        functional_test.go:66: 
            	Error Trace:	functional_test.go:66
            	            				then.go:54
            	            				then.go:31
            	            				functional_test.go:65
            	Error:      	Not equal: 
            	            	expected: "Error"
            	            	actual  : "Failed"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,2 +1,2 @@
            	            	-(v1alpha1.NodePhase) (len=5) "Error"
            	            	+(v1alpha1.NodePhase) (len=6) "Failed"
            	            	 
            	Test:       	TestFunctionalSuite/TestDeletingRunningPod
        functional_test.go:69: 
            	Error Trace:	functional_test.go:69
            	            				then.go:54
            	            				then.go:31
            	            				functional_test.go:65
            	Error:      	Not equal: 
            	            	expected: "pod deleted during operation"
            	            	actual  : "failed with exit code 2"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	-pod deleted during operation
            	            	+failed with exit code 2
            	Test:       	TestFunctionalSuite/TestDeletingRunningPod

```

--- a/test/e2e/functional_test.go
+++ b/test/e2e/functional_test.go
@@ -63,10 +63,15 @@ func (s *FunctionalSuite) TestDeletingRunningPod() {
 		WaitForWorkflow().
 		Then().
 		ExpectWorkflow(func(t *testing.T, metadata *metav1.ObjectMeta, status *wfv1.WorkflowStatus) {
-			assert.Equal(t, wfv1.NodeError, status.Phase)
-			assert.Len(t, status.Nodes, 1)
-			if assert.Contains(t, status.Nodes, metadata.Name) {
+			// the outcome could be either of these, depending on time
+			// this is due to the grace period recently deleted pods get
+			switch status.Phase {
+			case wfv1.NodeError:
 				assert.Equal(t, "pod deleted during operation", status.Nodes[metadata.Name].Message)
+			case wfv1.NodeFailed:
+				assert.Contains(t, status.Nodes[metadata.Name].Message, "failed with exit code")
+			default:
+				assert.Fail(t, "expected error of failed")
 			}
 		})
 }

